### PR TITLE
Merge credentials into proper location

### PR DESCRIPTION
**Pull request by mbonig**
Updates the way the .credentials are built to pull from the specific fields.

Addresses issue: https://github.com/chiefbiiko/dynamodb/issues/9

### CODE REVIEW

File: client/derive_config.ts [hunk 1]
```diff
@@ -17,7 +17,12 @@ export function deriveConfig(conf: ClientConfig = {}): Doc {
     const got: Doc = get({ profile: _conf.profile || "default" });
 
     if (typeof _conf.credentials !== "function") {
-      _conf.credentials = { ...got.credentials, ..._conf.credentials };
+      _conf.credentials = {
+        ...got.credentials,
+        ..._conf.credentials,
+        accessKeyId: got.accessKeyId,
+        secretAccessKey: got.secretAccessKey
+      };
```
Comment (mbonig) [inline client/derive_config.ts]: I believe we also need the SessionToken here too, agreed?

File: client/derive_config.ts [hunk 2]
```diff
@@ -17,7 +17,12 @@ export function deriveConfig(conf: ClientConfig = {}): Doc {
     const got: Doc = get({ profile: _conf.profile || "default" });
 
     if (typeof _conf.credentials !== "function") {
-      _conf.credentials = { ...got.credentials, ..._conf.credentials };
+      _conf.credentials = {
+        ...got.credentials,
```
Comment (mbonig) [inline client/derive_config.ts]: Should we purposely strip the accessKeyId and secretAccessKey off of this before spreading? It's polluting the object right now.

Comment (chiefbiiko) [reply client/derive_config.ts]: as you correctly said b4 `got` just doesn't have `credentials`

File: client/derive_config.ts (hunk 1 above)
Comment (chiefbiiko) [reply client/derive_config.ts]: yea, session token should be passed

**Comment by chiefbiiko** [comment]
thanks for this pr
i already added one commit bb3072819583897722eb99619cb99c1b2854b4bb that addresses this matter - sorry for the ignorance..
anyhow, i believe there is still sth wrong with `get-aws-config` - for me it has not been picking up `region` as it should

**Comment by mbonig** [comment]
> thanks for this pr
> i already added one commit [bb30728](https://github.com/chiefbiiko/dynamodb/commit/bb3072819583897722eb99619cb99c1b2854b4bb) that addresses this matter - sorry for the ignorance..
> anyhow, i believe there is still sth wrong with `get-aws-config` - for me it has not been picking up `region` as it should

Agreed, I had to manually override it.

File: client/derive_config.ts (hunk 2 above)
Comment (mbonig) [reply client/derive_config.ts]: 👍

**Comment by chiefbiiko** [comment]
Fixed in v0.3.0

**Pull request closed without merging**